### PR TITLE
[ET-947] DQA JS Update - Added dependency conditional support for notices

### PR DESCRIPTION
**Pull request by rafsuntaskin**
[ET-947]

Screen Capture on 2020-12-08 at 03-17-30

[ET-947]: https://moderntribe.atlassian.net/browse/ET-947

### CODE REVIEW

File: src/Tribe/Editor/Warnings.php [hunk 1]
```diff
@@ -106,11 +100,19 @@ public function get_recurring_event_warning_message() {
...
-		$icon = 'dashicons-' . $type;
+	public function render_notice( $message, $type = 'info', $depends_on = '', $condition = '' ) {
+		$icon           = 'dashicons-' . $type;
+		$has_dependency = empty( $depends_on ) ? '' : 'tribe-dependent';
+		$classes        = $type . ' ' . $has_dependency;
+		$condition_attr = empty( $condition ) ? '' : 'data-condition-is-' . $condition;
 		?>
-		<div class="ticket-editor-notice <?php echo esc_attr( $type ); ?>">
+		<div class="ticket-editor-notice <?php echo esc_attr( $classes ); ?>"
+		     data-depends="<?php echo esc_attr( $depends_on ); ?>"
```
Comment (tr1b0t) [inline src/Tribe/Editor/Warnings.php:113]: * :warning: Found precision alignment of 1 spaces.
* :no_entry_sign: Tabs must be used to indent lines; spaces are not allowed

<sub>hash: 2566f955001b262cd1fa736f6e9284a1</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 1 error(s)
:warning: 1 warning(s)

File: src/Tribe/Editor/Warnings.php (hunk 1 above)
Comment (sc0ttkclark) [inline src/Tribe/Editor/Warnings.php:113]:
```suggestion
			data-depends="<?php echo esc_attr( $depends_on ); ?>"
```

**Review by sc0ttkclark — CHANGES_REQUESTED** [review]
Slight tweak would be good here

**Review by sc0ttkclark — APPROVED** [review]

**Pull request merged**